src/avk.cpp

```diff
@@ -7101,11 +7103,78 @@ namespace avk
 		return create_semaphore(device(), dispatch_loader_core(), std::move(aAlterConfigBeforeCreation));
 	}
 
+	semaphore root::create_timeline_semaphore(uint64_t aPayload, std::function<void(semaphore_t&)> aAlterConfigBeforeCreation)
+	{
+		return create_semaphore(device(), dispatch_loader_core(), [otherAlterations = move(aAlterConfigBeforeCreation), aPayload](semaphore_t& aSem) {
+			auto typeInfo = std::make_unique<vk::SemaphoreTypeCreateInfo>();
+			typeInfo->semaphoreType = vk::SemaphoreType::eTimeline;
+			typeInfo->initialValue = aPayload;
+
+			aSem.create_info().pNext = typeInfo.get();
+			aSem.set_custom_deleter([aTypeInfo = move(typeInfo)]() { /* Do nothing ... this lambda just keeps the typeInfo struct alive */ });

```

Comment:
There's no point in keeping just the type info alive. We don't need to keep that alive because that does not represent a handle or something. Also if we put that into a custom deleter (which means inside a lambda), we lose access to it anyways. 

We could store it as a member of `semaphore_t`, if we still need the information stored. Do we? If not, just ignore.

Reply:
The reason I needed to keep the `typeInfo` alive was that it is created inside the lambda. Thus it would be destroyed once the lambda returned before the semaphore has even been created. Can be easily avoided by moving the `typeInfo` outside of the lambda and capturing it.

Reply:
Ah, I see. I think, you've already made the changes to have `typeInfo` outside of the lambda.
I like this approach better, because it does not rely on the lambda's lifetime at all, and keeps the lambda's as small as possible (size-wise based on captures).